### PR TITLE
Rebuild app navigation in JS

### LINKED ISSUE #8361: RouteNotFoundException once after app deactivation
I got following messages once after app deactivation (music and news have been disabled) It seems that the apps are loaded when they get deactivated. (looks like the navigation build step)

```
{"app":"index","message":"Symfony\\Component\\Routing\\Exception\\RouteNotFoundException: Route \"music_index\" does not exist.","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #1 \/srv\/http\/core\/lib\/private\/urlgenerator.php(41): OC\\Route\\Router->generate('music_index', Array)","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #2 \/srv\/http\/core\/lib\/private\/helper.php(44): OC\\URLGenerator->linkToRoute('music_index', Array)","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #3 \/srv\/http\/core\/lib\/public\/util.php(211): OC_Helper::linkToRoute('music_index', Array)","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #4 \/srv\/http\/core\/apps\/music\/appframework\/core\/api.php(239): OCP\\Util::linkToRoute('music_index', Array)","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #5 \/srv\/http\/core\/apps\/music\/appinfo\/app.php(41): OCA\\Music\\AppFramework\\Core\\API->linkToRoute('music_index')","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #6 \/srv\/http\/core\/lib\/private\/app.php(317): require('\/srv\/http\/core\/...')","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #7 \/srv\/http\/core\/settings\/ajax\/navigationdetect.php(9): OC_App::getAppNavigationEntries('music')","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #8 \/srv\/http\/core\/lib\/private\/route\/route.php(135) : runtime-created function(1): require_once('\/srv\/http\/core\/...')","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #9 [internal function]: __lambda_func(Array)","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #10 \/srv\/http\/core\/lib\/private\/route\/router.php(206): call_user_func('\\x00lambdabda_2, Array)","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #11 \/srv\/http\/core\/lib\/base.php(744): OC\\Route\\Router->match('\/settings\/ajax\/...')","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #12 \/srv\/http\/core\/index.php(28): OC::handleRequest()","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Exception: #13 {main}","level":4,"time":"2014-04-25T15:25:46+00:00"}
{"app":"index","message":"Symfony\\Component\\Routing\\Exception\\RouteNotFoundException: Route \"news.page.index\" does not exist.","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #1 \/srv\/http\/core\/lib\/private\/urlgenerator.php(41): OC\\Route\\Router->generate('news.page.index', Array)","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #2 \/srv\/http\/core\/lib\/private\/helper.php(44): OC\\URLGenerator->linkToRoute('news.page.index', Array)","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #3 \/srv\/http\/core\/lib\/public\/util.php(211): OC_Helper::linkToRoute('news.page.index', Array)","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #4 \/srv\/http\/news\/appinfo\/app.php(27): OCP\\Util::linkToRoute('news.page.index')","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #5 \/srv\/http\/core\/lib\/private\/app.php(317): require('\/srv\/http\/news\/...')","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #6 \/srv\/http\/core\/settings\/ajax\/navigationdetect.php(9): OC_App::getAppNavigationEntries('news')","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #7 \/srv\/http\/core\/lib\/private\/route\/route.php(135) : runtime-created function(1): require_once('\/srv\/http\/core\/...')","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #8 [internal function]: __lambda_func(Array)","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #9 \/srv\/http\/core\/lib\/private\/route\/router.php(206): call_user_func('\\x00lambdabda_2, Array)","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #10 \/srv\/http\/core\/lib\/base.php(744): OC\\Route\\Router->match('\/settings\/ajax\/...')","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #11 \/srv\/http\/core\/index.php(28): OC::handleRequest()","level":4,"time":"2014-04-25T15:28:46+00:00"}
{"app":"index","message":"Exception: #12 {main}","level":4,"time":"2014-04-25T15:28:46+00:00"}
```

### PR DESCRIPTION
**Pull request by RobinMcCorkell**
Retrieve all app navigations to prevent reloading appinfo/app.php and causing an error when the app isn't fully loaded. The addition/deletion logic has been moved to JS, simplifying a lot of code.

Fixes #8361

cc @MorrisJobke @PVince81

### CODE REVIEW

File: settings/ajax/navigationdetect.php [hunk 1]
```diff
@@ -23,14 +23,6 @@
 OC_Util::checkAdminUser();
 OCP\JSON::callCheck();
 
-$app = (string)$_GET['app'];
-$app = OC_App::cleanAppId($app);
+$navigation = \OC::$server->getNavigationManager()->getAll();
```
Comment (MorrisJobke) [inline settings/ajax/navigationdetect.php]: Use `OC_App::getNavigation()` to retrieve it as sorted list :see_no_evil:

**Comment by MorrisJobke** [comment]
Beside my comment this looks good :)

**Comment by Xenopathic** [comment]
@MorrisJobke @LukasReschke @bartv2 Review?

**Comment by LukasReschke** [comment]
Less PHP, more good :+1:

**Comment by PVince81** [comment]
Good idea. Needs rebase.
Also good luck writing JS unit tests :wink:

**Comment by PVince81** [comment]
Tested with the bookmarks app. Navigation entry is properly added/removed when enabling/disabling the app. :+1:

**Pull request merged**